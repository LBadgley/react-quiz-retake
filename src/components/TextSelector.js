import React from 'react';
import PropTypes from 'prop-types';
// need to get header and footer text from this to pass to Display
function TextSelector({ headerText, footerText, handleChange }) {
  return (
    <>
      <label htmlFor="headerText-input">Type something for the Header here: </label>
      <input name="header-text" type="text" value={headerText} id="headerText-input" onChange={handleChange} />
      <label htmlFor="footerText-input">Type something for the footer: </label>
      <input name="footer-text" type="text" value={footerText} id="footerText-input" onChange={handleChange} />
    </>
  );
}

TextSelector.propTypes = {
  headerText: PropTypes.string.isRequired,
  footerText: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default TextSelector;
