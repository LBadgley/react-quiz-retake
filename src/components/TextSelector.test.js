import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('Display Component text', () => {
  it('renders the display', () => {
    const wrapper = shallow(<TextSelector />);
    expect(wrapper).toMatchSnapshot();
  });
});
