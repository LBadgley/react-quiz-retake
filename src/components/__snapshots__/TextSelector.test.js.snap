// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Component text renders the display 1`] = `
<Fragment>
  <label
    htmlFor="headerText-input"
  >
    Type something for the Header here: 
  </label>
  <input
    id="headerText-input"
    name="header-text"
    type="text"
  />
  <label
    htmlFor="footerText-input"
  >
    Type something for the footer: 
  </label>
  <input
    id="footerText-input"
    name="footer-text"
    type="text"
  />
</Fragment>
`;
